.github/workflows/tests/test_autoscaling.py

```diff
@@ -0,0 +1,659 @@
+
+import json
+import os
+import subprocess
+import threading
+import time
+
+import pytest
+import requests
+
+
+def get_namespace():
+    """Get the target namespace from environment or default."""
+    return os.environ.get("NAMESPACE", "eoapi")
+
+
+def get_release_name():
+    """Get the release name from environment or default."""
+    return os.environ.get("RELEASE_NAME", "eoapi")
+
+
+def get_base_url():
+    """Get the base URL for API endpoints."""
+    # Try to detect ingress or use port-forward
+    namespace = get_namespace()
+
+    # Check if we have an ingress
+    result = subprocess.run(
+        ["kubectl", "get", "ingress", "-n", namespace, "-o", "json"],
+        capture_output=True,
+        text=True,
+    )
+
+    if result.returncode == 0:
+        ingress_data = json.loads(result.stdout)
+        if ingress_data["items"]:
+            ingress = ingress_data["items"][0]
+            rules = ingress.get("spec", {}).get("rules", [])
+            if rules:
+                host = rules[0].get("host", "localhost")
+                # Check if host is accessible
+                try:
+                    response = requests.get(
+                        f"http://{host}/stac/collections", timeout=5
+                    )
+                    if response.status_code == 200:
+                        return f"http://{host}"
+                except requests.RequestException:
+                    pass
+
+    # Fallback to localhost (assuming port-forward)
+    return "http://localhost:8080"
+
+
+def kubectl_get(resource, namespace=None, label_selector=None, output="json"):
+    """Execute kubectl get command with optional parameters."""
+    cmd = ["kubectl", "get", resource]
+
+    if namespace:
+        cmd.extend(["-n", namespace])
+
+    if label_selector:
+        cmd.extend(["-l", label_selector])
+
+    if output:
+        cmd.extend(["-o", output])
+
+    result = subprocess.run(cmd, capture_output=True, text=True)
+    return result
+
+
+def get_pod_metrics(namespace, service_name):
+    """Get current CPU and memory metrics for service pods."""
+    release_name = get_release_name()
+    result = subprocess.run(
+        [
+            "kubectl",
+            "top",
+            "pods",
+            "-n",
+            namespace,
+            "-l",
+            f"app={release_name}-{service_name}",
+            "--no-headers",
+        ],
+        capture_output=True,
+        text=True,
+    )
+
+    if result.returncode != 0:
+        return []
+
+    metrics = []
+    for line in result.stdout.strip().split("\n"):
+        if line.strip():
+            parts = line.split()
+            if len(parts) >= 3:
+                pod_name = parts[0]
+                cpu = parts[1]  # e.g., "25m"
+                memory = parts[2]  # e.g., "128Mi"
+                metrics.append({"pod": pod_name, "cpu": cpu, "memory": memory})
+
+    return metrics
+
+
+def get_hpa_status(namespace, hpa_name):
```

Comment:
I would like to see typing of function parameters, but that's a personal pref...

Reply:
Added.